app/src/main/java/com/celzero/bravedns/net/go/GoVpnAdapter.java

```diff
@@ -205,37 +202,33 @@ private boolean isDoh(long dnsMode) {
         return (Settings.DNSModeIP == dnsMode || Settings.DNSModePort == dnsMode);
     }
 
-    private void setTunnelMode(long dnsMode, long blockMode, long proxyMode) {
-        if (!isDnscrypt(dnsMode)) {
-            if (isOrbotProxy(proxyMode)) {
-                tunnel.setTunMode(dnsMode, blockMode, Settings.ProxyModeSOCKS5);
+    private void setTunnelMode(AppMode.TunnelMode tunnelMode) {
+        if (!isDnscrypt(tunnelMode.getDnsMode())) {
```

Comment:
Move `isDnsCrypt`, `isOrbotProxy`, `isDnsProxy` and related functions into `TunnelOptions` itself.

Reply:
moved.